ompi/mca/osc/ucx/osc_ucx_component.c

```diff
@@ -45,7 +45,12 @@ ompi_osc_ucx_component_t mca_osc_ucx_component = {
         .osc_query = component_query,
         .osc_select = component_select,
         .osc_finalize = component_finalize,
-    }
+    },
+    .ucp_context            = NULL,
+    .ucp_worker             = NULL,
+    .env_initialized        = false,
+    .num_incomplete_req_ops = 0,
+    .init_in_progress       = 1
```

Comment:
why isn't mca_osc_ucx_component.num_incomplete_req_ops>0 ?

Reply:
to avoid branch

Reply:
actually one counter is not enough - we can't know when initialization is complete, and assuming that init complete when at least one operation is complete

Reply:
i mean, why when there is lack of progress, mca_osc_ucx_component.num_incomplete_req_ops is 0?
maybe we should call worker_progress in osc/ucs once in X times regardless of num_incomplete_req_ops?

Reply:
right, it doesn't progress till at least one operation is scheduled, and on startup worker progress is not called at all. I played with number of forced calls of progress, but this solution is not robust (how many calls should be done?) and makes code more complex.

I'm not sure if this solution is correct: on many ranks it may hang too (I tested 4x32 ranks only), the most visible fix is remove num_incomplete_req_ops evaluation from progress callback

Reply:
can we progress only if, for example, there is at least one window?

Reply:
it seems there is no existing global counters of windows, I can add it into UCX, but it will take time to test. and as I can see in most of cases in app where RMA is used at least one windows exists from start to finish of application's life

Reply:
@hoopoepg can we add such counter to osc_ucx?
especially if we have RMA over UD/TCP, would need progress on RX side all the time

Reply:
@yosefe ok, done